src/libs/workspace-utils.ts

```diff
@@ -128,14 +155,36 @@ export const isGoogleWorkspace = (workspace: BaseWorkspace): workspace is Google
 export const getCloudProviderFromWorkspace = (workspace: BaseWorkspace): CloudProvider =>
   isAzureWorkspace(workspace) ? cloudProviderTypes.AZURE : cloudProviderTypes.GCP;
 
-export const hasProtectedData = (workspace: BaseWorkspace): boolean => containsProtectedDataPolicy(workspace.policies);
+/**
+ * Determine whether a workspace is considered protected.
+ *
+ * For Azure workspaces, this checks for the "protected-data" policy.
+ * For Google workspaces, this checks for has enhanced logging - either directly or from an auth domain.
+ *
+ * @param workspace - The workspace.
+ */
+export const isProtectedWorkspace = (workspace: WorkspaceWrapper): boolean => {
+  switch (workspace.workspace.cloudPlatform) {
+    case 'Azure':
+      return containsProtectedDataPolicy(workspace.policies);
+    case 'Gcp':
+      return workspace.workspace.bucketName.startsWith('fc-secure');
+    default:
+      // Check that all possible cases are handled.
+      const exhaustiveGuard: never = workspace.workspace;
+      return exhaustiveGuard;
```

Comment:
I've seen this before and was wondering if this was the generally accepted way to express exhaustiveness. Does it have a different meaning if the variable is inlined?
```suggestion
      return workspace.workspace as never;
```

Reply:
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking